src/filters/array.rs

```diff
@@ -0,0 +1,217 @@
+        .as_array()
+        .ok_or_else(|| invalid_input("Array expected"))?
+        .iter();
+
+    let mut sentence = array
+        .next()
+        .and_then(|v| Some(v.to_string()))
+        .unwrap_or_else(|| "".to_string());
+
+    let last = array.next_back();
+
+    for value in array {
+        write!(sentence, ", {}", value).unwrap();
+    }
+
+    if let Some(last) = last {
+        write!(sentence, " {} {}", connector, last).unwrap();
+    }
+
+    Ok(Value::scalar(sentence))
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    macro_rules! unit {
+        ($a:ident, $b:expr) => {{
+            unit!($a, $b, &[])
+        }};
+        ($a:ident, $b:expr, $c:expr) => {{
+            $a(&$b, $c).unwrap()
+        }};
+    }
+
+    #[test]
+    fn unit_push() {
+        let input = Value::Array(vec![Value::scalar("Seattle"), Value::scalar("Tacoma")]);
+        let args = &[Value::scalar("Spokane")];
+        let desired_result = Value::Array(vec![
+            Value::scalar("Seattle"),
+            Value::scalar("Tacoma"),
+            Value::scalar("Spokane"),
+        ]);
+        assert_eq!(unit!(push, input, args), desired_result);
+    }
+
+    #[test]
+    fn unit_pop() {
+        let input = Value::Array(vec![Value::scalar("Seattle"), Value::scalar("Tacoma")]);
+        let args = &[];
+        let desired_result = Value::Array(vec![Value::scalar("Seattle")]);
+        assert_eq!(unit!(pop, input, args), desired_result);
+    }
+
+    #[test]
+    fn unit_pop_empty() {
+        let input = Value::Array(vec![]);
+        let args = &[];
+        let desired_result = Value::Array(vec![]);
+        assert_eq!(unit!(pop, input, args), desired_result);
+    }
+
+    #[test]
+    fn unit_unshift() {
+        let input = Value::Array(vec![Value::scalar("Seattle"), Value::scalar("Tacoma")]);
+        let args = &[Value::scalar("Olympia")];
+        let desired_result = Value::Array(vec![
+            Value::scalar("Olympia"),
+            Value::scalar("Seattle"),
+            Value::scalar("Tacoma"),
+        ]);
+        assert_eq!(unit!(unshift, input, args), desired_result);
+    }
+
+    #[test]
+    fn unit_shift() {
+        let input = Value::Array(vec![Value::scalar("Seattle"), Value::scalar("Tacoma")]);
+        let args = &[];
+        let desired_result = Value::Array(vec![Value::scalar("Tacoma")]);
+        assert_eq!(unit!(shift, input, args), desired_result);
+    }
+
+    #[test]
+    fn unit_shift_empty() {
+        let input = Value::Array(vec![]);
+        let args = &[];
+        let desired_result = Value::Array(vec![]);
+        assert_eq!(unit!(shift, input, args), desired_result);
+    }
+
+    #[test]
+    fn unit_array_to_sentence_string() {
+        let input = Value::Array(vec![
+            Value::scalar("foo"),
+            Value::scalar("bar"),
+            Value::scalar("baz"),
+        ]);
+        let args = &[];
+        let desired_result = Value::scalar("foo, bar and baz");
```

Comment:
1. Does jekyll's version leave off the oxford comma?

With oxford comma

> foo, bar, and baz

Without

> foo, bar and baz

2. Should we follow them?

Reply:
I was about to say that they use no oxford comma, but then I went back and they do, in fact.

I don't know how that went over me, I really didn't notice it and I had looked at it at least a dozen of times.

Reply:
Very easy to miss., I obviously missed it the first time I reviewed the code and I was looking at tests which helps.